CMakeLists.txt

```diff
@@ -914,6 +905,51 @@ else()
         set_source_files_properties(${default_plugin_sources} ${ua_architecture_sources} PROPERTIES LANGUAGE CXX)
     endif()
 
+    # Declare include directories
+    function(add_private_build_include_directories)
```

Comment:
I would go for a bit shorter name: `include_directories_public` and `include_directories_private`. This is also more like the original CMake name.